airflow-core/src/airflow/api_fastapi/core_api/routes/public/xcom.py

```diff
@@ -193,8 +194,10 @@ def get_xcom_entries(
     query = query.order_by(
         XComModel.dag_id, XComModel.task_id, XComModel.run_id, XComModel.map_index, XComModel.key
     )
-    xcoms = session.scalars(query)
-    return XComCollectionResponse(xcom_entries=xcoms, total_entries=total_entries)
+    xcoms = session.scalars(query).all()
+    return XComCollectionResponse(
+        xcom_entries=[XComResponse.from_orm(xcom) for xcom in xcoms], total_entries=total_entries
```

Comment:
Will MyPy raise error if we just use `session.scalars(query)` ?
```suggestion
        xcom_entries=session.scalars(query), total_entries=total_entries
```